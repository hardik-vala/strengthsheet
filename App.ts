import { StatusBar } from 'expo-status-bar';
import { StyleSheet, Text, TextInput, View } from 'react-native';

export default function App() {
  return (
    <View style={styles.container}>
      <Text>Start working on your app!</Text>
      <StatusBar style="auto" />
      <TextInput
        style={{
          height: 40,
          width: '80%',
          borderColor: 'gray',
          borderWidth: 1,
        }}
        defaultValue="You can type in me"
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
