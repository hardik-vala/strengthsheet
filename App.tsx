import {
  GoogleSignin,
  GoogleSigninButton,
} from "@react-native-google-signin/google-signin";
import { StatusBar } from "expo-status-bar";
import { useState } from "react";
import { Alert, Button, StyleSheet, Text, TextInput, View } from "react-native";
import { signIn, signOut } from "./authService";
import { User } from "./models/User";

export default function App() {
  const [user, setUser] = useState<User>(null);
  const [userInput, setUserInput] = useState("");

  const _signIn = async () => {
    try {
      const userInfo = await signIn();
      setUser(userInfo);
    } catch (error) {
        Alert.alert(error);
    }
  };

  const _signOut = async () => {
    try {
      signOut();
      setUser(null);
    } catch (error) {
      Alert.alert(error);
    }
  };

  function render() {
    const body = user ? renderMain() : renderSignInButton();
    return body;
  }

  function renderMain() {
    return (
      <>
        <Text>Enter a workout value:</Text>
        <StatusBar style="auto" />
        <TextInput
          style={{
            height: 40,
            width: "80%",
            borderColor: "gray",
            borderWidth: 1,
          }}
          defaultValue=""
          placeholder="Type here"
          value={userInput}
          onChangeText={(text) => setUserInput(text)}
        />
        <Button title="Submit" onPress={submitToSheet} />
        <Button title="Log out" onPress={_signOut} />
      </>
    );
  }

  function renderSignInButton() {
    return (
      <>
        <GoogleSigninButton
          size={GoogleSigninButton.Size.Standard}
          color={GoogleSigninButton.Color.Dark}
          onPress={_signIn}
        />
      </>
    );
  }

  async function submitToSheet() {
    const authTokens = await GoogleSignin.getTokens();
    const spreadsheetId = "1-wL-dRJYZkZ-uVpoBSuGeSFEzg_ZWVKFwLTv8RgbX7o";

    const url = `https://sheets.googleapis.com/v4/spreadsheets/${spreadsheetId}/values/Sheet1!A1:A1:append?valueInputOption=RAW`;

    const headers = {
      Authorization: `Bearer ${authTokens.accessToken}`,
    };

    const body = {
      range: "Sheet1!A1:A1",
      majorDimension: "ROWS",
      values: [[userInput]],
    };

    const response = await fetch(url, {
      method: "POST",
      headers,
      body: JSON.stringify(body),
    });

    if (!response.ok) {
      console.error(`Submit to sheet failed: ${JSON.stringify(response)}`);
    }
  }

  return <View style={styles.container}>{render()}</View>;
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
});
