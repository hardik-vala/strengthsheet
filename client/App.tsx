import { useState } from "react";
import { Alert, StyleSheet, View } from "react-native";
import { MD3LightTheme, PaperProvider } from "react-native-paper";
import { SafeAreaProvider } from "react-native-safe-area-context";
import { SignIn } from "./src/components/SignIn/SignIn";
import { WorkoutPortal } from "./src/components/WorkoutPortal/WorkoutPortal";
import { WORKOUT_TEMPLATE_REGISTRY } from "./src/data/registry";
import { User } from "./src/models/User";
import { signIn, signOut } from "./src/services/authService";

const theme = {
  ...MD3LightTheme,
  colors: {
    ...MD3LightTheme.colors,
    primary: "black",
    secondary: "gray",
  },
};

export default function App() {
  const [user, setUser] = useState<User>(null);

  const _signIn = async () => {
    try {
      const user = await signIn();
      setUser(user);
    } catch (error) {
      Alert.alert(error);
    }
  };

  const _signOut = async () => {
    try {
      signOut();
      setUser(null);
    } catch (error) {
      Alert.alert(error);
    }
  };

  return (
    <SafeAreaProvider>
      <PaperProvider theme={theme}>
        <View style={styles.container}>
          <Content
            user={user}
            onSignIn={_signIn}
            onSignOut={_signOut}
          ></Content>
        </View>
      </PaperProvider>
    </SafeAreaProvider>
  );
}

export function Content({ user, onSignIn, onSignOut }) {
  if (user) {
    return (
      <WorkoutPortal
        workoutTemplateRegistry={WORKOUT_TEMPLATE_REGISTRY}
        onSignOut={onSignOut}
      />
    );
  }

  return <SignIn onSignIn={onSignIn} />;
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    justifyContent: "center",
  },
});
