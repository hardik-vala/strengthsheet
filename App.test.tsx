import { render } from "@testing-library/react-native";
import React from "react";

import { Content } from "./App";

jest.mock("./src/components/Main/Main", () => {
  return {
    Main: jest.fn(() => <div>Mock Main</div>),
  };
});

jest.mock("./src/components/SignIn/SignIn", () => {
  return {
    SignIn: jest.fn(() => <div>Mock SignIn</div>),
  };
});

jest.mock("./src/services/authService", () => {
  return {
    signIn: jest.fn(),
    signOut: jest.fn(),
  };
});

const mockOnSignIn = jest.fn();
const mockOnSignOut = jest.fn();

describe("Content", () => {
  it("renders sign in if the user hasn't signed in", () => {
    const tree = render(<Content user={null} onSignIn={mockOnSignIn} onSignOut={mockOnSignOut}/>).toJSON();

    expect(tree.children.length).toBe(1);
    expect(tree.children[0]).toEqual("Mock SignIn");
  });

  it("renders main page if the user has signed in", () => {
    const tree = render(<Content user={{}} onSignIn={mockOnSignIn} onSignOut={mockOnSignOut}/>).toJSON();

    expect(tree.children.length).toBe(1);
    expect(tree.children[0]).toEqual("Mock Main");
  });
});
