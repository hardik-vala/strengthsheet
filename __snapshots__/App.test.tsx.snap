// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders main page if the user has signed in 1`] = `
<div>
  Mock Main
</div>
`;

exports[`Content renders sign in if the user hasn't signed in 1`] = `
<div>
  Mock SignIn
</div>
`;
